homeassistant/components/hydrawise/binary_sensor.py

```diff
@@ -15,22 +18,40 @@
 from .coordinator import HydrawiseDataUpdateCoordinator
 from .entity import HydrawiseEntity
 
-BINARY_SENSOR_STATUS = BinarySensorEntityDescription(
-    key="status",
-    device_class=BinarySensorDeviceClass.CONNECTIVITY,
+
+@dataclass(frozen=True, kw_only=True)
+class HydrawiseBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes Hydrawise binary sensor."""
+
+    value_fn: Callable[[HydrawiseBinarySensor], bool | None]
+
+
+CONTROLLER_BINARY_SENSORS: tuple[HydrawiseBinarySensorEntityDescription, ...] = (
+    HydrawiseBinarySensorEntityDescription(
+        key="status",
+        device_class=BinarySensorDeviceClass.CONNECTIVITY,
+        value_fn=lambda status_sensor: status_sensor.coordinator.last_update_success,
+    ),
 )
 
-BINARY_SENSOR_TYPES: tuple[BinarySensorEntityDescription, ...] = (
-    BinarySensorEntityDescription(
-        key="is_watering",
-        translation_key="watering",
+RAIN_SENSOR_BINARY_SENSOR: tuple[HydrawiseBinarySensorEntityDescription, ...] = (
+    HydrawiseBinarySensorEntityDescription(
+        key="rain_sensor",
+        translation_key="rain_sensor",
         device_class=BinarySensorDeviceClass.MOISTURE,
+        value_fn=lambda rain_sensor: rain_sensor.sensor.status.active,
     ),
 )
 
-BINARY_SENSOR_KEYS: list[str] = [
-    desc.key for desc in (BINARY_SENSOR_STATUS, *BINARY_SENSOR_TYPES)
-]
+ZONE_BINARY_SENSORS: tuple[HydrawiseBinarySensorEntityDescription, ...] = (
+    HydrawiseBinarySensorEntityDescription(
+        key="is_watering",
+        translation_key="watering",
+        device_class=BinarySensorDeviceClass.RUNNING,
+        value_fn=lambda watering_sensor: watering_sensor.zone.scheduled_runs.current_run
```

Comment:
Please use parenthesis around the lambda and indent it so that it's easier to distinguish the lambda from the other entity description parameters.